src/js/firstparties/facebook.js

```diff
@@ -0,0 +1,63 @@
+// Adapted from https://github.com/mgziminsky/FacebookTrackingRemoval
+(function() {
+let fb_wrapped_link = "a[href*='facebook.com/l.php?'";
+
+function findInAllFrames(query) {
+  let out = [];
+  document.querySelectorAll(query).forEach((node) => {
+    out.push(node);
+  });
+  Array.from(document.getElementsByTagName('iframe')).forEach((iframe) => {
+    try {
+      iframe.contentDocument.querySelectorAll(query).forEach((node) => {
+        out.push(node);
+      });
+    } catch (e) {
+      // pass on cross origin iframe errors
+    }
+  });
+  return out;
+}
+
+// remove all attributes from a link except for class and ARIA attributes
+function cleanAttrs(elem) {
+  for (let i = elem.attributes.length - 1; i >= 0; --i) {
+    const attr = elem.attributes[i];
+    if (attr.name !== 'class' && !attr.name.startsWith('aria-')) {
+      elem.removeAttribute(attr.name);
+    }
+  }
+}
+
+// Remove excessive attributes and event listeners from link a and replace
+// its destination with href
+function cleanLink(a) {
+  let href = new URL(a.href).searchParams.get('u');
+  cleanAttrs(a);
+  a.href = href;
+  a.target = "_blank";
```

Comment:
Why set `target="_blank"`? (Isn't it already present?) Should we also set [`rel="noopener"`](https://developers.google.com/web/tools/lighthouse/audits/noopener)?

Reply:
We [set `rel="noreferrer"` on Twitter](https://github.com/EFForg/privacybadger/blob/558ecb5481c504d31e043b2270ad4e2323c2908e/src/js/firstparties/twitter.js#L5).